### ISSUE TITLE
error in catchment delineation, TypeError: Iterator operand 0 dtype could not be cast from dtype('float64') to dtype('int64') according to the rule 'same_kind'

### ISSUE THREAD

**Issue by ShashankBice** [mdbartos/pysheds#47]
Hi @mdbartos,
I use the following command :- 
dirmap = (64, 128, 1, 2, 4, 8, 16, 32)
x,y = 664976.60,5315794.1
grid.catchment(data='dir',x=x,y=y,dirmap=dirmap,out_name='catch',nodata_in=grid.nodata,nodata_out=grid.nodata)
and get the error:-
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-56-f4e4ad7c905b> in <module>
      1 dirmap = (64, 128, 1, 2, 4, 8, 16, 32)
      2 x,y = 664976.60,5315794.1
----> 3 grid.catchment(data='dir',x=x,y=y,dirmap=dirmap,out_name='catch',nodata_in=grid.nodata,nodata_out=grid.nodata)

~/.conda/envs/bhushanPy3/lib/python3.6/site-packages/pysheds-0.2.2-py3.6.egg/pysheds/grid.py in catchment(self, x, y, data, pour_value, out_name, dirmap, nodata_in, nodata_out, xytype, routing, recursionlimit, inplace, apply_mask, ignore_metadata, **kwargs)
    856                                       xytype=xytype, recursionlimit=recursionlimit, inplace=inplace,
    857                                       apply_mask=apply_mask, ignore_metadata=ignore_metadata,
--> 858                                       properties=properties, metadata=metadata, **kwargs)
    859         elif routing.lower() == 'dinf':
    860             return self._dinf_catchment(x, y, fdir=fdir, pour_value=pour_value, out_name=out_name,

~/.conda/envs/bhushanPy3/lib/python3.6/site-packages/pysheds-0.2.2-py3.6.egg/pysheds/grid.py in _d8_catchment(self, x, y, fdir, pour_value, out_name, dirmap, nodata_in, nodata_out, xytype, recursionlimit, inplace, apply_mask, ignore_metadata, properties, metadata, **kwargs)
    893             # get the flattened index of the pour point
    894             pour_point = np.ravel_multi_index(np.array([y, x]),
--> 895                                               fdir.shape)
    896             # reorder direction mapping to work with select_surround_ravel()
    897             r_dirmap = np.array(dirmap)[[4, 5, 6, 7, 0, 1, 2, 3]].tolist()

TypeError: Iterator operand 0 dtype could not be cast from dtype('float64') to dtype('int64') according to the rule 'same_kind'

**Comment by mdbartos** [comment]
Specify that you are searching for x,y coordinates rather than x,y indices by using the xytype keyword argument:

```python
grid.catchment(data='dir',x=x,y=y,dirmap=dirmap, out_name='catch', xytype='label')
```

Right now it thinks you are looking for the array row (5315794.1) and column (664976.60).

Also, you don't need to specify the nodata values in the keyword arguments. It will use the nodata value associated with that dataset by default.

**Comment by mdbartos** [comment]
This should be fixed in v0.3 due to the strict typing required by numba. Please reopen if the issue persists.
